lib/company.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const Promise = require('bluebird')
+
+/**
+ * Company - Used for communication with intercomClient's company object
+ * @class
+ * @author Ryan Kahn
+ */
+class Company {
+
+  /**
+   * Constructor for Company
+   * @param {Orion} orion - Instance of Orion which is used for the intercom client and canUseIntercom methods
+   */
+  constructor (orion) {
+    this.orion = orion
+    if (!this.orion.canUseIntercom()) { return }
```

Comment:
want to add a check to ensure `this.orion.canUseIntercom` is a function before calling it? it will throw regardless but you might be able to add a better error message. also since you defined a class you can ensure the right `typeof` is passed in :)